dallinger/experiment_server/experiment_server.py

```diff
@@ -296,29 +294,36 @@ def advertisement():
     except exc.SQLAlchemyError:
         status = None
 
-    if status == STARTED and not debug_mode:
-        # Once participants have finished the instructions, we do not allow
-        # them to start the task again.
-        raise ExperimentError('already_started_exp_mturk')
-    elif status == COMPLETED:
+    if status == 'working' and part.end_time is not None:
```

Comment:
That's not the case. I tried this at first, before I completely understood how the notifications were working. `'submitted'` means that `externalSubmit` has been called and the HIT is complete. A failure callback could still be received from mturk after `worker_complete`.

`worker_complete` is called in the popup window, the job of which is essentially to mark the user's session as being over. It then refreshes its `window.opener` and closes itself. The `window.opener` is the iframe inside mechanical turk, which then needs to show the `externalSubmit` button, which actually ends the task. As soon as `externalSubmit` is called, the parent of that iframe, the HIT info page on MTurk changes to being a confirmed page and a callback is made to set the status to be `'submitted'`. If the iframe is reloaded before the completed state it shows an error.

One other way of doing this would be to change the state enum to have a state between `'working'` and `'submitted'`, like `'readytosubmit'` and have `worker_complete` update the status there.